app/scripts/modules/titus/src/pipeline/stages/runJob/TitusRunJobStageConfig.tsx

```diff
@@ -0,0 +1,383 @@
+  };
+
+  private updateImageId() {
+    const { stage } = this.props;
+    if (stage.repository && stage.tag) {
+      stage.cluster.imageId = `${stage.repository}:${stage.tag}`;
+    } else {
+      delete stage.cluster.imageId;
+    }
+  }
+
+  private dockerChanged = (changes: IDockerImageAndTagChanges) => {
+    Object.assign(this.props.stage, changes);
+    this.updateImageId();
+    this.props.stageFieldUpdated();
+    this.setState({});
+  };
+
+  private stageFieldChanged = (fieldIndex: string, value: any) => {
+    set(this.props.stage, fieldIndex, value);
+    this.props.stageFieldUpdated();
+    this.setState({});
+  };
+
+  public componentDidMount() {
+    const { application, stage } = this.props;
+    stage.cluster = stage.cluster || {};
+    stage.waitForCompletion = stage.waitForCompletion === undefined ? true : stage.waitForCompletion;
+
+    if (stage.cluster.imageId) {
+      const image = stage.cluster.imageId;
+      stage.organization = '';
+      const parts = image.split('/');
+      if (parts.length > 1) {
+        stage.organization = parts.shift();
+      }
+
+      const rest = parts.shift().split(':');
+      if (stage.organization) {
+        stage.repository = `${stage.organization}/${rest.shift()}`;
+      } else {
+        stage.repository = rest.shift();
+      }
+      stage.tag = rest.shift();
+    }
+
+    if (!stage.credentials && application.defaultCredentials.titus) {
+      stage.credentials = application.defaultCredentials.titus;
+    }
+
+    if (!stage.cluster.capacity) {
+      stage.cluster.capacity = {
+        min: 1,
+        max: 1,
+        desired: 1,
+      };
+    }
+
+    stage.cloudProvider = stage.cloudProvider || 'titus';
+    stage.cluster.application = stage.cluster.application || application.name;
+    stage.cluster.env = stage.cluster.env || {};
+    stage.cluster.resources = stage.cluster.resources || {};
+    stage.cluster.resources.cpu = stage.cluster.resources.cpu || 1;
+    stage.cluster.resources.disk = stage.cluster.resources.disk || 10000;
+    stage.cluster.resources.gpu = stage.cluster.resources.gpu || 0;
+    stage.cluster.resources.memory = stage.cluster.resources.memory || 512;
+    stage.cluster.resources.networkMbps = stage.cluster.resources.networkMbps || 128;
+    stage.cluster.retries = stage.cluster.retries || 0;
+    stage.cluster.runtimeLimitSecs = stage.cluster.runtimeLimitSecs || 3600;
+    stage.cluster.securityGroups = stage.cluster.securityGroups || [];
+    stage.deferredInitialization = true;
+
+    $q.all({
+      credentialsKeyedByAccount: AccountService.getCredentialsKeyedByAccount('titus'),
+    }).then(backingData => {
+      this.credentialsKeyedByAccount = backingData.credentialsKeyedByAccount;
+      const credentials = Object.keys(backingData.credentialsKeyedByAccount);
+
+      if (!stage.credentials) {
+        stage.credentials = credentials[0];
+      }
+
+      this.setRegistry(stage.credentials);
+
+      return $q.all([]).then(() => {
```

Comment:
???

Reply:
No idea, just left it there, will remove.